File: schedule-service/src/main/java/pe/albrugroup/schedule_service/entity/request/horario/RegistrarExcepcionHorarioRequest.java
package pe.albrugroup.schedule_service.entity.request.horario;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import pe.albrugroup.schedule_service.entity.enums.TipoExcepcionHorario;

import java.time.LocalDate;
import java.time.LocalTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RegistrarExcepcionHorarioRequest {

    @NotNull(message = "fecha es obligatoria")
    private LocalDate fecha;

    @NotNull(message = "tipo es obligatorio")
    private TipoExcepcionHorario tipo;

    private LocalTime horaEntrada;
    private LocalTime horaSalida;
    private LocalTime inicioAlmuerzo;
    private LocalTime finAlmuerzo;
    private Boolean laborable;

    @NotBlank(message = "motivo es obligatorio")
    private String motivo;
}

File: frontend/src/app/shared/models/schedule/registrar-excepcion-horario-request.ts
export interface RegistrarExcepcionHorarioRequest {
  fecha: string;
  tipo: string;
  horaEntrada?: string | null;
  horaSalida?: string | null;
  inicioAlmuerzo?: string | null;
  finAlmuerzo?: string | null;
  laborable?: boolean | null;
  motivo: string;
}
